### PR TITLE
[tmclient] Migrate flags to pflag

### LINKED ISSUE #10987: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tmclient`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tmclient/*.go
var TabletManagerProtocol = flag.String("tablet_manager_protocol", "grpc", "the protocol to use to talk to vttablet")
```

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This PR migrates the `tmclient` flags (just `--tablet_manager_protocol`) to `pflag` and scopes them to the appropriate binaries.

Here's every entry point that imports:

```
go/cmd/mysqlctl => nope, only starts/stops mysql
go/cmd/mysqlctld => nope, same as above
go/cmd/rulesctl => nope
    go/cmd/rulesctl/cmd
    go/cmd/rulesctl/common
go/cmd/vtadmin => nope
go/cmd/vtbackup => yep, needs it
go/cmd/vtcombo => yep
go/cmd/vtctl
go/cmd/vtctld
go/cmd/vtctldclient
go/cmd/vtexplain => unsure, gonna rely on tests for this one (probably not needed though)
go/cmd/vtgate => unsure, relying on tests, but doesn't seem needed
go/cmd/vtgateclienttest => if vtgate is "no", this is probably also "no"
go/cmd/vtgr => yep, needed
go/cmd/vtorc => yep
go/cmd/vttablet => yep
go/cmd/vttestserver => yep
go/cmd/zk => nope
```

## Related Issue(s)

Closes #10987.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/flags/endtoend/vtctld.txt [hunk 1]
```diff
@@ -217,7 +217,7 @@ Usage of vtctld:
       --tablet_manager_grpc_crl string                                   the server crl to use to validate server certificates when connecting
       --tablet_manager_grpc_key string                                   the key to use to connect
       --tablet_manager_grpc_server_name string                           the server name to use to validate server certificate
-      --tablet_manager_protocol string                                   the protocol to use to talk to vttablet (default "grpc")
+      --tablet_manager_protocol string                                   Protocol to use to talk to vttablets. (default "grpc")
```
Comment (ajm188) [inline go/flags/endtoend/vtctld.txt:220]: Taking suggestions on better help text to distinguish between this and `--tablet_protocol` (this was kinda already a problem, I'm just making it even more obvious)

**Review by GuptaManan100 — COMMENTED** [review]
It looks good to me, but I am not sure if vtgate also needs the `tablet_manager_protocol` flag. It would be nice if @harshit-gangal @systay or @frouioui could also take a look

**Comment by ajm188** [comment]
My read on this is that vtgate only uses `queryservice` RPCs, which are governed by `tabletconn` and the `--tablet_protocol` flag, but let me know if that's not correct!

**Comment by harshit-gangal** [comment]
`tablet_protocol` is needed for vtgate to know which registered service to use.

File: go/flags/endtoend/vtctld.txt (hunk 1 above)
Comment (harshit-gangal) [reply go/flags/endtoend/vtctld.txt]: > tablet_manager_protocol

AFAIK, VTGate does not use `tablet_manager_protocol`

**Comment by harshit-gangal** [comment]
> It looks good to me, but I am not sure if vtgate also needs the `tablet_manager_protocol` flag. It would be nice if @harshit-gangal @systay or @frouioui could also take a look

AFAIK, VTGate does not use `tablet_manager_protocol`

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**